### ISSUE TITLE
When serializing objects weaved with the GoAOP framework, some data should be excluded by default

### ISSUE THREAD

**Issue by Rastusik** [schmittjoh/serializer#854]
| Q                | A
| ---------------- | -----
| Bug report?      | no
| Feature request? | yes
| BC Break report? | no
| RFC?             | no

## Steps required to reproduce the problem

1. Weave an object using GoAOP (http://go-aop-php.readthedocs.io/en/dev/articles/limitations_and_known_issues/serialization-reflections-and-weaving.html)
2. try to serialize it

## Expected Result

Only serialized data from the object nothing else.

## Actual Result

Serialied data from the object + the data from the `$__joinPoints` variable used by the proxies generated by GoAOP.

**Comment by goetas** [comment]
cant this be achieved with `@ExclusionPolicy("ALL")` ?

**Comment by goetas** [comment]
in alternative, a listener similar to https://github.com/schmittjoh/serializer/blob/master/src/JMS/Serializer/EventDispatcher/Subscriber/DoctrineProxySubscriber.php could be used

**Comment by Rastusik** [comment]
yes, bouth solutions are okey, but with the exlusion policy, one has to map every other fields of the object ... I don't really understand the subscriber solution, but it looks like a better approach. The question is, would this be available in the serializer package, or are you suggesting to create my own subscriber in my own app?

**Comment by goetas** [comment]
it can be included in the serializer package in the same way how is available doctrine. i've never worked with GoAOP, so definitively will not be able to help (except of reviewing the code..) and since i've never worked with GoAOP do not have plans to work on a feature for it.

The trick in the listener approach is [this on line 87](https://github.com/schmittjoh/serializer/blob/master/src/JMS/Serializer/EventDispatcher/Subscriber/DoctrineProxySubscriber.php#L87), basically detects the type looking at the parent... all the rest of the code are "optimizations"

**Comment by Rastusik** [comment]
@goetas yes, I was able to find the trick. I can provide the subscriber when this becomes a show stopper for us, right know we were able to workaround the whole situation. But still thanks for the disclaimer!
